src/arch/arm/insts/static_inst.cc

```diff
@@ -752,17 +752,17 @@ ArmStaticInst::checkAdvSIMDOrFPEnabled32(ThreadContext *tc,
 
     if (have_virtualization && !is_secure) {
         HCPTR hcptr = tc->readMiscReg(MISCREG_HCPTR);
-        bool hcptr_cp10 = hcptr.tcp10;
+        bool hcptr_tcp10 = hcptr.tcp10;
         bool hcptr_tase = hcptr.tase;
 
         if (have_security && !ELIs64(tc, EL3) && !is_secure) {
             if (nsacr.nsasedis)
                 hcptr_tase = true;
-            if (nsacr.cp10)
```

Comment:
I feel like this RAO/RAZ handling should really be done inside readMiscReg() in the same way other registers handle it but for now I just submitted the minimal fix.

Reply:
Understand your point, on the other hand, this logic is closely matching the pseudocode in the arm architectural reference manual.. Lots of time we have sacrificed the best implementation on gem5 for the sake of being amenable to debug/extend whenever extensions are added